homeassistant/components/zwave_js/update.py

```diff
@@ -0,0 +1,190 @@
+from zwave_js_server.client import Client as ZwaveClient
+from zwave_js_server.const import NodeStatus
+from zwave_js_server.exceptions import BaseZwaveJSServerError
+from zwave_js_server.model.driver import Driver
+from zwave_js_server.model.firmware import FirmwareUpdateInfo
+from zwave_js_server.model.node import Node as ZwaveNode
+
+from homeassistant.components.update import UpdateDeviceClass, UpdateEntity
+from homeassistant.components.update.const import UpdateEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import API_KEY_FIRMWARE_UPDATE_SERVICE, DATA_CLIENT, DOMAIN, LOGGER
+from .helpers import get_device_id, get_valueless_base_unique_id
+
+PARALLEL_UPDATES = 1
+SCAN_INTERVAL = timedelta(days=1)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Z-Wave button from config entry."""
+    client: ZwaveClient = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+
+    @callback
+    def async_add_firmware_update_entity(node: ZwaveNode) -> None:
+        """Add firmware update entity."""
+        driver = client.driver
+        assert driver is not None  # Driver is ready before platforms are loaded.
+        async_add_entities([ZWaveNodeFirmwareUpdate(driver, node)], True)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            f"{DOMAIN}_{config_entry.entry_id}_add_firmware_update_entity",
+            async_add_firmware_update_entity,
+        )
+    )
+
+
+class ZWaveNodeFirmwareUpdate(UpdateEntity):
+    """Representation of a firmware update entity."""
+
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_device_class = UpdateDeviceClass.FIRMWARE
+    _attr_supported_features = (
+        UpdateEntityFeature.INSTALL | UpdateEntityFeature.RELEASE_NOTES
+    )
+    _attr_has_entity_name = True
+
+    def __init__(self, driver: Driver, node: ZwaveNode) -> None:
+        """Initialize a Z-Wave device firmware update entity."""
+        self.driver = driver
+        self.node = node
+        self.available_firmware_updates: list[FirmwareUpdateInfo] = []
+        self._latest_version_firmware: FirmwareUpdateInfo | None = None
+        self._status_unsub: Callable[[], None] | None = None
+
+        # Entity class attributes
+        self._attr_name = "Firmware"
+        self._base_unique_id = get_valueless_base_unique_id(driver, node)
+        self._attr_unique_id = f"{self._base_unique_id}.firmware_update"
+        # device may not be precreated in main handler yet
+        self._attr_device_info = DeviceInfo(
```

Comment:
For a different PR: I'm surprised there is no standardized way for this. Now you need to keep this in sync for every platform.

Reply:
it's only for these one off device level entities we create. I started trying to fix this but it created other problems that I wasn't ready to solve